src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -249,14 +708,20 @@ def map_keyvalue_to_featureflag(keyvalue, show_conditions=True):
     if filters and state == FeatureState.ON:
         state = FeatureState.CONDITIONAL
```

Comment:
What is our strategy for handling "conditional" state? Is it still controlled by the presence of conditions only? What if the state is enabled but it has variants?

Reply:
Currently we just have it as it was if there are filters and state of the feature flag is enabled then the state is conditional. Should we have it as conditional if there are variants?

Reply:
The concept of `state `seems to only exist in the CLI and its only used for display. It makes sense to me to have it consistent with portal as well.

Reply:
I agree with getting rid of the `state` property altogether in exchange for the boolean `enabled`. This will then eliminate any confusion that the `conditional` state currently causes.

Reply:
This would technically be a breaking change if someone has a dependency on the output field "state". But I don't think we can avoid that. The current output makes no sense in context of variants. Maybe we can call it out in the release notes?

Reply:
I think calling it out in the release notes is a good idea. I would like to suggest having this in another PR and release it during the breaking change window.

Reply:
When is the next breaking change release planned?
If we dont make the change here, I'm not sure what the alternative is for "State" property with variants.

Reply:
I think the breaking change would around May. I am thinking we could have the state as 
`"Variant-based"`. Still open to other suggestions. What do you think? Although what if a flag has both variants and filters, what should we show..

Reply:
It is a bit confusing. 
Maybe we can keep the output with "State" if compatibility mode is set to true. If compatibility is false, it means users have explicitly opted for v2 schema and we can show a different output format. The new format can show the same fields we display in the portal. And once we make V2 schema default, we can fully adopt the new output.

If we agree on that, we can keep showing "Conditional" state whenever a flag is not explicitly enabled or disabled. So any combination of conditions/variants/allocation will make the flag conditional.